client/src/main/java/io/split/client/LocalhostSplitClientAndFactory.java

```diff
@@ -0,0 +1,72 @@
+package io.split.client;
+
+import io.split.client.api.Key;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * An implementation of SplitClient that considers all partitions
+ * passed in the constructor to be 100% on for all users, and
+ * any other split to be 100% off for all users. This implementation
+ * is useful for using Split in localhost environment.
+ *
+ * @author adil
+ */
+public final class LocalhostSplitClientAndFactory implements SplitClient {
+    private static final Logger _log = LoggerFactory.getLogger(LocalhostSplitClientAndFactory.class);
+
+    private LocalhostSplitFactory _factory;
+    private LocalhostSplitClient _splitClient;
+
+    public LocalhostSplitClientAndFactory(LocalhostSplitFactory container, LocalhostSplitClient client) {
+        _factory = container;
+        _splitClient = client;
+
+        checkNotNull(_factory);
+        checkNotNull(_splitClient);
+    }
+
+    @Override
+    public String getTreatment(String key, String split) {
+        return _splitClient.getTreatment(key, split);
+    }
+
+    @Override
+    public String getTreatment(String key, String split, Map<String, Object> attributes) {
+        return _splitClient.getTreatment(key, split, attributes);
+    }
+
+    @Override
+    public String getTreatment(Key key, String split, Map<String, Object> attributes) {
```

Comment:
why are these methods are and also in the client?

Reply:
because this is a composite class meant to pull out the factory from the client. 

Before my changes, the LocalhostSplitClient had a reference to the factory that created it. That seems odd. Plus it makes unit testing harder. So I pulled the factory out of the client and made a composite class that could delegate to the client for client work and destroy the factory when it needs to.